src/libcrun/utils.c

```diff
@@ -1916,35 +1930,57 @@ find_executable (const char *executable_path, const char *cwd)
     {
       ret = check_access (executable_path);
       if (ret == 0)
-        return xstrdup (executable_path);
-      return NULL;
+        {
+          *out = xstrdup (executable_path);
+          return 0;
+        }
     }
 
   end = tmp = xstrdup (getenv ("PATH"));
 
   while ((it = strsep (&end, ":")))
     {
-      size_t len;
+      cleanup_free char *path = NULL;
 
       if (it == end)
         it = ".";
 
-      len = snprintf (path, PATH_MAX, "%s/%s", it, executable_path);
-      if (len >= PATH_MAX)
-        continue;
+      ret = append_paths (&path, err, it, executable_path, NULL);
+      if (UNLIKELY (ret < 0))
+        {
+          crun_error_release (err);
+          continue;
+        }
 
       ret = check_access (path);
       if (ret == 0)
-        return xstrdup (path);
+        {
+          /* Change owner.  */
+          *out = path;
+          path = NULL;
+          return 0;
+        }
 
-      if (errno == ENOENT)
+      if (ret == -ENOENT)
         continue;
 
-      last_error = errno;
+      last_error = ret;
     }
 
-  errno = last_error;
-  return NULL;
+  switch (last_error)
+    {
+    case 1:
```

Comment:
would it make sense to use a define for 1 and 2, reading the code here alone it was not clear to me where  the meaning of 1 and 2 is defined